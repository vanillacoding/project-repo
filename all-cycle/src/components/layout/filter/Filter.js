import {
  Container,
  Name,
  ToggleButton,
} from "@/components/layout/filter/styled";

function Filter({
  name,
  TYPES,
  sortFilter,
  sortWithTypes,
}) {
  return (
    <Container>
      <Name>{name}</Name>

      {TYPES.map((type) => (
        <ToggleButton
          key={JSON.stringify(type)}
          isclicked={sortFilter.productType === type[0]}
          onClick={() => sortWithTypes("productType", type[0])}
          size={0.4}
        >
          {type[1]}
        </ToggleButton>
      ))}
    </Container>
  );
}

export default Filter;
