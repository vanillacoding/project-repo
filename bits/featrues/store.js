import { configureStore } from '@reduxjs/toolkit';
import ReduxThunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { userSlice } from './userSlice';
import { habitSlice } from './habitSlice';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistStore, persistReducer } from 'redux-persist';

const userPersistConfig = {
  key: 'user',
  storage: AsyncStorage,
  blacklist: ['isSuccess', 'isError', 'isFetching']
};

const habitPersistConfig = {
  key: 'habit',
  storage: AsyncStorage,
  blacklist: ['isSuccess', 'isError', 'isFetching']
};

const reducer = {
  user: persistReducer(userPersistConfig, userSlice.reducer),
  habit: persistReducer(habitPersistConfig, habitSlice.reducer)
};

const middleware = [ReduxThunk, createLogger()];

const store = configureStore({
  reducer,
  middleware
});

const persistedStore = persistStore(store);

export { store, persistedStore };
