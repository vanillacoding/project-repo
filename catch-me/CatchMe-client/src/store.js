import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger"

import single from "./redux/slices/singlePlaySlice"; 
import multiple from "./redux/slices/multiplaySlice";
import socketMiddleware from "./redux/middleware/socket";

const store = configureStore({
  reducer: {
    single,
    multiple,
  },
  middleware: (getDefaultMiddleware) => {
    if (process.env.NODE_ENV !== "production") {
      return getDefaultMiddleware().concat([logger, socketMiddleware()]);
    }

    return getDefaultMiddleware().concat([logger, socketMiddleware()]);
  },
  devTools: process.env.NODE_ENV !== "production",
});

export default store;
