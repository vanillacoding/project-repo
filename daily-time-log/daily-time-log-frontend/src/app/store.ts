import { configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import logger from "redux-logger";

import authReducer from "../features/auth/authSlice";
import calendarReducer from "../features/calendar/calendarSlice";
import schedulesReducer from "../features/schedules/schedulesSlice";
import milestonesReducer from "../features/milestones/milestonesSlice";
import timeLogReducer from "../features/timeLog/timeLogSlice";
import settingReducer from "../features/setting/settingSlice";

const store = configureStore({
  reducer: {
    auth: authReducer,
    calendar: calendarReducer,
    schedules: schedulesReducer,
    milestones: milestonesReducer,
    timeLog: timeLogReducer,
    setting: settingReducer,
  },
  middleware: (getDefaultMiddleware) => {
    return process.env.REACT_APP_NODE_ENV !== "production"
      ? getDefaultMiddleware().concat(logger)
      : getDefaultMiddleware();
  },
  devTools: process.env.REACT_APP_NODE_ENV !== "production",
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
