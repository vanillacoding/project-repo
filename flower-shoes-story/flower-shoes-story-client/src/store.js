import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import storage from "redux-persist/lib/storage";

import userReducer from "./features/userSlice";
import preloadReducer from "./features/preloadSlice";

const reducers = combineReducers({
  user: userReducer,
  preload: preloadReducer,
});

const persistConfig = {
  key: "root",
  storage,
  whitelist: ["user"],
};

const persistedReducer = persistReducer(persistConfig, reducers);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [logger],
});

export default store;
