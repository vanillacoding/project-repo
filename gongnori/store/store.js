import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { createLogger } from "redux-logger";
import { NODE_ENV } from "@env";
import reducer from "../reducers/reducer";

const middleware = [thunk];

if (NODE_ENV === "development") {
  middleware.push(createLogger());
}

export default store = createStore(reducer, applyMiddleware(...middleware));
