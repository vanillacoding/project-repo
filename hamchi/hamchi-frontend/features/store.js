import { configureStore } from '@reduxjs/toolkit';
import ReduxThunk from 'redux-thunk';
import { userSlice } from './userSlice';
import { postSlice } from './postSlice';
import { chatSlice } from './chatSlice';
import { filteredPostSlice } from './filteredPostSlice';
import { createLogger } from 'redux-logger';

const middleware = [ReduxThunk];

const store = configureStore({
  reducer:
  {
    user: userSlice.reducer,
    post: postSlice.reducer,
    chat: chatSlice.reducer,
    filteredPost: filteredPostSlice.reducer
  },
  middleware: middleware
});

export default store;
