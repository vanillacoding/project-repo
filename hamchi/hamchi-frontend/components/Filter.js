import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addType, deleteType, initFeeds } from '../features/filteredPostSlice';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

import enumToString from '../constants/mapEnumToString';
import hamsterTypes from '../constants/hamsterTypes';
import colors from '../theme/color';

const Filter = ({ title }) => {
  const dispatch = useDispatch();
  const selectedHamsterTypes = useSelector(state => state.filteredPost.selectedHamsterTypes);
  const mapped = enumToString.hamsterType;

  function handleSelectType(type) {
    dispatch(initFeeds(handleSelectType));

    if (selectedHamsterTypes[type]) {
      dispatch(deleteType(type));
    } else {
      dispatch(addType(type));
    }
  }

  return (
    <View style={styles.container}>
      <Text style={styles.text}>{title}</Text>
      <View style={styles.tagContainer}>
        {hamsterTypes.map(type => {
          return (
            <TouchableOpacity
              style={[
                styles.tag,
                { backgroundColor: selectedHamsterTypes[type] ? colors.main : colors.white }
              ]}
              key={type}
              onPress={() => handleSelectType(type)}
            >
              <Text style={[
                styles.text,
                { color: selectedHamsterTypes[type] ? colors.white : colors.main }
              ]}>{mapped[type]}</Text>
            </TouchableOpacity>
          );
        })}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 10,
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: colors.white,
    marginBottom: 0
  },
  tagContainer: {
    flexDirection: "row",
  },
  tag: {
    marginTop: 12,
    width: 50,
    height: 24,
    margin: 5,
    marginBottom: -20,
    flexDirection: "row",
    justifyContent: "center",
    borderRadius: 6,
  },
  text: {
    fontWeight: "bold",
    alignSelf: "center",
  }
});

export default Filter;
