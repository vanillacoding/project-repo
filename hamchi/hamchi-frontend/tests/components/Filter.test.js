import React from 'react';
import { Provider } from 'react-redux';
import store from '../../features/store';
import renderer from 'react-test-renderer';
import Filter from '../../components/Filter';

describe('filter test', () => {
  it('renders correctly', () => {
    const mockTitle = 'mock title';
    const tree = renderer.create(
      <Provider store={store}>
        <Filter title={mockTitle} />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
