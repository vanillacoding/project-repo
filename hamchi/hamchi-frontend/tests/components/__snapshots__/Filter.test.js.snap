// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexDirection": "column",
      "margin": 10,
      "marginBottom": 0,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontWeight": "bold",
      }
    }
  >
    mock title
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 6,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "margin": 5,
          "marginBottom": -20,
          "marginTop": 12,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontWeight": "bold",
            },
            Object {
              "color": "#F48D3B",
            },
          ]
        }
      >
        골든
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 6,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "margin": 5,
          "marginBottom": -20,
          "marginTop": 12,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontWeight": "bold",
            },
            Object {
              "color": "#F48D3B",
            },
          ]
        }
      >
        드워프
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 6,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "margin": 5,
          "marginBottom": -20,
          "marginTop": 12,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontWeight": "bold",
            },
            Object {
              "color": "#F48D3B",
            },
          ]
        }
      >
        로보
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 6,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "margin": 5,
          "marginBottom": -20,
          "marginTop": 12,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "fontWeight": "bold",
            },
            Object {
              "color": "#F48D3B",
            },
          ]
        }
      >
        그 외
      </Text>
    </View>
  </View>
</View>
`;
